packages/astro/src/core/logger/vite.ts

```diff
@@ -0,0 +1,88 @@
+import { fileURLToPath } from 'url';
+import stripAnsi from 'strip-ansi';
+import type { Logger as ViteLogger, Rollup, LogLevel } from 'vite';
+import { isAstroError } from '../errors/errors.js';
+import { isLogLevelEnabled, type Logger as AstroLogger } from './core.js';
+
+const PKG_PREFIX = fileURLToPath(new URL('../../../', import.meta.url));
+const E2E_PREFIX = fileURLToPath(new URL('../../../e2e', import.meta.url));
+export function isAstroSrcFile(id: string | null) {
+	return id?.startsWith(PKG_PREFIX) && !id.startsWith(E2E_PREFIX);
+}
+
+// capture "page reload some/Component.vue (additional info)" messages
+const vitePageReloadMsg = /page reload (.*)( \(.*\))?/;
+// capture "hmr update some/Component.vue" messages
+const viteHmrUpdateMsg = /hmr update (.*)/;
+// capture "vite v5.0.0 building SSR bundle for production..." and "vite v5.0.0 building for production..." messages
+const viteBuildMsg = /vite.*building.*for production/;
+
+export function createViteLogger(
+	astroLogger: AstroLogger,
+	viteLogLevel: LogLevel = 'info'
+): ViteLogger {
+	const warnedMessages = new Set<string>();
+	const loggedErrors = new WeakSet<Error | Rollup.RollupError>();
+
+	const logger: ViteLogger = {
+		hasWarned: false,
+		info(msg) {
+			if (!isLogLevelEnabled(viteLogLevel, 'info')) return;
+
+			const stripped = stripAnsi(msg);
+			let m;
+			// Rewrite HMR page reload message
+			if ((m = vitePageReloadMsg.exec(stripped))) {
```

Comment:
The fact that some files are logged by Vite with a leading `/` and some are not is pretty odd and would confuse me as a user (especially on windows where I imagine one would have `/` and the other would have `\`). Does Vite consider this a bug?

I also don't love matching strings like this with regex, since these error messages aren't usually covered by semver and can change unexpectedly across different versions.

I think I prefer the solution that keeps handling this ourselves in `handleHotUpdate` where we can control things ourselves, for those reasons. But I also could be missing some benefits!

Reply:
I've left a note in the description about the slash issue:

> NOTE: there's a leading slash inconcistency for [watch] messages, seems to be intended in Vite. Leading slash is the URL, non-leading slash is for local files.

I investigated that yesterday and it seems safer this way since you may sometime get `/@fs/some/fs/path` and it would not be safe to remove the initial slash.

If we put back in `handleHotUpdate`, we won't be able to fix https://github.com/withastro/astro/issues/8224 though. Since an update that calls `handleHotUpdate` doesn't always trigger HMR, e.g. if you're editing files that are not part of your app.